roles/azure_manage_snapshot/meta/argument_specs.yml

```diff
@@ -0,0 +1,34 @@
+---
+argument_specs:
+  main:
+    version_added: 3.1.0
+    short_description: A role to manage a snapshot to an Azure Virtual Machine.
+    description:
+      - A role to manage a snapshot to an Azure Virtual Machine.
+      - This role requires an azure user account with valid permission.
+    options:
+      azure_manage_snapshot_operation:
+        description:
+          - Operation to perform
+        default: "create"
+        choices: ["create", "delete", "restore"]
+      azure_manage_snapshot_resource_group:
+        description:
+          - Resource group from which the snapshot resides.
+        required: true
+      azure_manage_snapshot_name:
+        description:
+          - Name of the snapshot to restore.
+        required: true
+      azure_manage_snapshot_vm_name:
+        description:
+          - Name of the virtual machine to restore the snapshot to.
+        required: true
```

Comment:
It is not required for delete operation

Reply:
really wish we had required_if in argument spec.  I'll change this to false.